Makefile

```diff
@@ -47,10 +50,70 @@ push: $(TARGETS:=.push)
 release: $(TARGETS:=.release)
 	@true
 
-clean:
-	rm -rf $(OUTPUT_DIR)
-
 license:
 	licenser apply -r . "Tetrate"
 
+
 .PHONY: clean push release license
+
+.PHONY: api
+api: api/manifest.proto $(BUF) $(PROTOC_GEN_GO)
+	@echo "--- $@: api ---"
+	@rm api/*.go
+	@$(BUF) protoc \
+		--plugin=protoc-gen-go=$(PROTOC_GEN_GO) \
+		--go_out=paths=source_relative:. \
+		api/manifest.proto
+
+LINT_OPTS ?= --timeout 5m
+.PHONY: lint
+lint: $(GOLANGCI_LINT) $(SHFMT) $(LICENSER) .golangci.yml  ## Run the linters
+	@echo "--- lint ---"
+	@$(SHFMT) -d .
+	@$(LICENSER) verify -r .
+	@$(GOLANGCI_LINT) run $(LINT_OPTS) --config .golangci.yml
+
+# The goimports tool does not arrange imports in 3 blocks if there are already more than three blocks.
+# To avoid that, before running it, we collapse all imports in one block, then run the formatter.
+.PHONY: format
+format: $(GOIMPORTS) $(SHFMT) ## Format all Go code
+	@echo "--- format ---"
+	@$(SHFMT) -w .
+	@$(LICENSER) apply -r "Tetrate"
+	@find . -type f -name '*.go' | xargs gofmt -s -w
+	@for f in `find . -name '*.go'`; do \
+	    awk '/^import \($$/,/^\)$$/{if($$0=="")next}{print}' $$f > /tmp/fmt; \
+	    mv /tmp/fmt $$f; \
+	    $(GOIMPORTS) -w -local github.com/tetratelabs/getenvoy-package $$f; \
+	done
+
+# Enforce go version matches what's in go.mod when running `make check` assuming the following:
+# * 'go version' returns output like "go version go1.16 darwin/amd64"
+# * go.mod contains a line like "go 1.16"
+EXPECTED_GO_VERSION_PREFIX := "go version go$(shell sed -ne '/^go /s/.* //gp' go.mod )"
+GO_VERSION := $(shell go version)
+
+.PHONY: check
```

Comment:
this is the new pre-flight check which is similar to what's happening in https://github.com/tetratelabs/getenvoy except this has a protobuf (api) step